### DISCUSSION TITLE
How are you handling relations?

### DISCUSSION THREAD

**Discussion by ItsRobbAllen** [ravendb/ravendb#17113] [General]
We recently had a synchronization issue from our legacy system where someone made several changes directly to the SQL database, bypassing the API, which lead to a bunch of records in our Raven database being orphaned. Our synchronization process removes orphaned records but those documents were referenced elsewhere in the system and it forced me to do a rather painful, manual update in several places.

So, my question to all is what patterns are you all using to manage referential integrity? We store 'stub' information in many classes simply to provide the id and maybe the name/description of the 'reference' and _most_ of the time this is a display-only issue & not a big deal if it gets out of sync.

However, when the data is actually important—for ex. a store has a reference to the company that owns it—and something changes that is critical to spread throughout the system, what do you do to keep things in sync?

Just looking at general ideas/concepts that others are currently doing and if anyone has any comments as to things to look out for, etc.

We do a TON of ETLs between microservice databases and 99% of the time, the ids do not change so this isn't generally anything more than a display issue, but that 1% was a little painful :/

**Comment by ayende** [comment]
The key problem for RavenDB with regards to relations is that it is distributed multi master.
So you can create an `Order` for `Customer` on one node and have another node delete that `Customer`.

Ways to avoid that include using cluster wide transactions that would ensure consistency (but obviously more complex) or setting up a "detect orphans" index.
